src/components/Article.jsx

```diff
@@ -32,6 +36,22 @@ export default function Article({ article }) {
     return created_at.split("T")[0];
   }
 
+  function handleVote(num) {
+    setVotes((prev) => prev + num);
+    const updatedVotes = votes + num;
+    setArticleWithBody((prevArticle) => ({
+      ...prevArticle,
+      votes: updatedVotes,
+    }));
+
+    patchArticle(article_id, num)
+      .then((updatedArticle) => setVotes(updatedArticle.votes))
+      .catch((err) => {
+        setVotes(votes);
+
+        setVoteError(err.message);
+      });
```

Comment:
Should also reverse the optimistic vote here, so that if my vote fails the votes end up the same